tutorials/create-openwrt-image-for-scaleway/index.mdx

```diff
@@ -0,0 +1,343 @@
+- Edit the image to fit Scaleway's ecosystem
+- Upload the image to Scaleway Object Storage
+- Convert to Snapshot through import
+- Test the image and troubleshoot via Console
+
+<Message type="requirement">
+
+  - You have an account and are logged into the [Scaleway Console](https://console.scaleway.com/)
+  - You have [generated an SSH Key](/console/my-project/how-to/create-ssh-key/)
+  - You have [created](/storage/object/how-to/create-a-bucket/) a Scaleway Object Storage Bucket
+
+</Message>
+
+The following commands are done on an Ubuntu 22.04.
+
+## Download the image
+
+The needed image must be a full disk image, not an ISO image or only a rootfs.
+
+In order to work on Scaleway Instances, the image must be using EFI (not just BIOS) to boot.
+
+OpenWRT images are available [here](https://openwrt.org/downloads), we are going to use a stable release, for `x86_64` architecture, with EFI.
+
+Download the image we are going to use in this tutorial:
+
+  ```
+  curl -sSLO https://downloads.openwrt.org/releases/22.03.2/targets/x86/64/openwrt-22.03.2-x86-64-generic-ext4-combined-efi.img.gz
+  gunzip openwrt-22.03.2-x86-64-generic-ext4-combined-efi.img.gz
+  ```
+
+## Prepare the QCOW2 image
+
+<Message type="note">
+
+QCOW2 images are container images used in [QEMU](https://www.qemu.org/) as virtual disks.
+
+</Message>
+
+1. Install the needed QEMU tools:
+
+    ```
+    apt-get install qemu-utils -y
+    ```
+
+2. Convert the image and resize the target disk size (minimum disk size at Scaleway = 1GB):
+
+    ```
+    qemu-img convert -f raw -O qcow2 openwrt-22.03.2-x86-64-generic-ext4-combined-efi.img openwrt-22.03.2-x86-64-generic-ext4-combined-efi.qcow2
+    qemu-img resize openwrt-22.03.2-x86-64-generic-ext4-combined-efi.qcow2 1G
+    ```
+
+3. Mount the QCOW2 image as a device
+
+    ```
+    modprobe nbd
+    qemu-nbd -c /dev/nbd0 openwrt-22.03.2-x86-64-generic-ext4-combined-efi.qcow2
+    ```
+
+    <Message type="important">
+
+    Conversion and resizing creates errors in the partition table which are fixed when re-writing it using fdisk. Plus, to resize the main partition, the partition number must be found out.
+
+    To print and fix partition table, do:
+
+    ```
+    echo 'p\nw\n' | fdisk /dev/nbd0
+    ```
+
+    </Message>
+
+4. Resize the main partition to fit the available space (partition number 2 in the example above):
+
+    ```
+    growpart /dev/nbd0 2
+    resize2fs /dev/nbd0p2
+    ```
+
+## Edit the image content
+
+<Message type="note">
+
+Network configuration and specifics actions may need to be configured in order to access the image when run in Scaleway Instances.
+
+For example:
+
+* In Scaleway Instances, the first NIC (e.g. `eth0`) must be associated to the public network and need to be configured (For exapmple using: Cloud-init, DHCP, ...)
+
+* If the image provides a GUI, it may need to be configured to allow access from public interface (Are specific ports or firewall rules required?)
```

Comment:
```suggestion
* If the image provides a GUI, it may need to be configured to allow access from public interface (are specific ports or firewall rules required?)
```